components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/reply/CorrelationTimeoutMap.java

```diff
@@ -16,97 +16,58 @@
  */
 package org.apache.camel.component.rabbitmq.reply;
 
-import java.util.concurrent.ScheduledExecutorService;
-
-import org.apache.camel.support.DefaultTimeoutMap;
+import org.apache.camel.TimeoutMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * A {@link org.apache.camel.TimeoutMap} which is used to track reply messages which
  * has been timed out, and thus should trigger the waiting {@link org.apache.camel.Exchange} to
  * timeout as well.
  */
-public class CorrelationTimeoutMap extends DefaultTimeoutMap<String, ReplyHandler> {
+public class CorrelationTimeoutMap implements TimeoutMap<String, ReplyHandler> {
```

Comment:
Its a bit map in map in map ;). Just wonder if we could avoid all the delegate/wrapping, and let it be so you can extend DefaultTimeoutMap as before